core/commands/storage/session.go

```diff
@@ -191,9 +198,24 @@ func NewSessionID() (string, error) {
 	return ssid.String(), nil
 }
 
+func GetFileMetaFromDatabase(node *core.IpfsNode, ssID string) (*FileContracts, error) {
+	rds := node.Repo.Datastore()
+	value, err := rds.Get(ds.NewKey(FileContractsStorePrefix + ssID))
```

Comment:
Actually in order to reuse the same code for host, I would suggest to pass Prefix as parameter, and rename ssID in this function to key, so it can be more generalized.